### DISCUSSION TITLE
Test Report: Pixel 3A XL / bonito is functional and relockable

### DISCUSSION THREAD

**Discussion by ZhanYF** [Divested-Mobile/DivestOS-Build#272] [General]
I'm glad to report that Pixel 3A XL works with the official DivestOS build and the bootloader can be relocked. I followed the fastboot installation guide and it worked as expected. I then relocked the bootloader and now it displays the correct AVB hash during boot time. 

Tested hardware features:
- camera
- wifi
- speaker
- fingerprint reader

Thanks! Let me know if you want me to test a specific feature.

**Comment by SkewedZeppelin** [comment]
Thank you for testing & reporting, I'll mark it working.

**Comment by jefe276** [comment]
Hey, trying to get Divest on 3a XL - can't get the Recovery working, just reboots. Any advice?

**Reply by SkewedZeppelin** [reply]
What system were you running previously?

The `fastboot update divested-20.0-20240513-dos-bonito-fastboot.zip` step completed successfully?

**Comment by jefe276** [comment]
I reinstalled the 9.0 factory image, because the fastboot wouldn't install without it and it was recommended somewhere in the documentation. Fastboot.zip provided installs, but rebooting to Recovery doesn't work. Curious why "if recovery.IMG is available" step is included if there is no 3a XL image provided by Divest...

Sent from Proton Mail Android

-------- Original Message --------

**Reply by SkewedZeppelin** [reply]
No where does DivestOS recommend flashing Android 9 firmware for bonito.

The steps are right here https://divestos.org/builds/LineageOS/bonito/install.html

9.0 is likely way too old and so the outdated firmware is incompatible

try flash-base.sh from here first: https://dl.google.com/dl/android/aosp/bonito-sp2a.220505.008-factory-db19d2aa.zip

**Comment by jefe276** [comment]
Thanks for the advice.

I did try flashing with the latest factory image from 05052022, but then the fastboot.zip provided wouldn't flash at all - error due to some partitioning as I recall. Forum suggested the 9.0/version, which worked for flashing the fastboot, but again, I was unable to reboot that to Recovery/fastboot.

I've successfully put Divest on an old Nexus 6p and a Pixel 3, just can't manage my 3a XL, daily phone.

Sent from Proton Mail Android

-------- Original Message --------

**Reply by SkewedZeppelin** [reply]
Are you using fastboot version 33.0.3? newer versions will throw weird partition issues like that
https://divestos.org/pages/bootloader#prereqComputer

also be sure you have enough free space in /tmp as the fastboot zip gets extracted there

**Comment by jefe276** [comment]
I was NOT using that fastboot version - I think you got it. Thanks for taking the time to straighten me out.

Sent from Proton Mail Android

-------- Original Message --------

**Comment by jefe276** [comment]
Hey, thanks again.

Was using Minimal ADB and FB instead of the platform tools version. Installed, bootloader locked and functioning.

Sent with [Proton Mail](https://proton.me/) secure email.

**Reply by SkewedZeppelin** [reply]
awesome :slightly_smiling_face:

**Comment by jefe276** [comment]
Hey, I didn't realize when I messaged you that you were actually The Guy from DivestOS.

As a former Grapheme guy, I'm digging Divest the more I'm nreading about it, especially since support from Grapheme has fallen off for Pixel 3a XL and I'm committed to making my devices last longer.

Quick question: vacationing in an area with spotty coverage, I triedWiFIi Calling - obviously it doesn't exist and I understand why, as it exposes actual IP to cell provider.

But when fiddling, the OS is stuck on LTE Only under preferred provider, and won't switch back to LTE preferred (recommended). Maybe a tweak to allow that flexibility to change that back.

Sent from Proton Mail Android

-------- Original Message --------

**Reply by SkewedZeppelin** [reply]
the lte only thing is just cosmetic, the actual selected option is what is set: https://github.com/Divested-Mobile/DivestOS-Build/issues/305

**Comment by jefe276** [comment]
Sorry to bother. If you have time....

Put this on a Nexus 6p - it's won't recognize the SIM card, make calls, texts, data, etc, because of that.

Put the factory image and a couple other ROMs on there, they DO recognize the SIM when it's in the Nexus 6P. Thoughts?

Sent from Proton Mail Android

-------- Original Message --------

**Reply by SkewedZeppelin** [reply]
Can you try running the flash radio scripts from the factory images again?
and also sideload the latest divestos again too

if it still doesn't work after please see if there are any errors from:
- `adb logcat -b all -d | grep -i -e radio -e modem -e network -e connect -e phone -e tele`
- `adb logcat -b all -d *:E`
- `adb logcat -b all -d | grep -i -e hardened_malloc -e "fatal allocator error"`
